analyzer/pkg/rpm/rpm.go

```diff
@@ -62,21 +67,63 @@ func (a rpmPkgAnalyzer) parsePkgInfo(packageBytes []byte) (pkgs []types.Package,
 	for _, pkg := range pkgList {
 		arch := pkg.Arch
 		if arch == "" {
-			arch = "(none)"
+			arch = "None"
 		}
+
+		// parse source rpm
+		var srcName, srcVer, srcRel string
+		if pkg.SourceRpm != "(none)" && pkg.SourceRpm != "" {
+			// source epoch is not included in SOURCERPM
+			srcName, srcVer, srcRel, _, _ = splitFileName(pkg.SourceRpm)
+		}
+
 		p := types.Package{
-			Name:    pkg.Name,
-			Epoch:   pkg.Epoch,
-			Version: pkg.Version,
-			Release: pkg.Release,
-			Arch:    arch,
+			Name:       pkg.Name,
+			Epoch:      pkg.Epoch,
+			Version:    pkg.Version,
+			Release:    pkg.Release,
+			Arch:       arch,
+			SrcName:    srcName,
+			SrcEpoch:   pkg.Epoch, // NOTE: use epoch of binary package as epoch of src package
+			SrcVersion: srcVer,
+			SrcRelease: srcRel,
 		}
 		pkgs = append(pkgs, p)
 	}
 
 	return pkgs, nil
 }
 
+// splitFileName returns a name, version, release, epoch, arch
+// e.g.
+//    foo-1.0-1.i386.rpm returns foo, 1.0, 1, i386
+//    1:bar-9-123a.ia64.rpm returns bar, 9, 123a, 1, ia64
+// https://github.com/rpm-software-management/yum/blob/043e869b08126c1b24e392f809c9f6871344c60d/rpmUtils/miscutils.py#L301
+func splitFileName(filename string) (name, ver, rel string, epoch int, arch string) {
+	if strings.HasSuffix(filename, ".rpm") {
+		filename = filename[:len(filename)-4]
+	}
+
+	archIndex := strings.LastIndex(filename, ".")
+	arch = filename[archIndex+1:]
+
+	relIndex := strings.LastIndex(filename[:archIndex], "-")
+	rel = filename[relIndex+1 : archIndex]
+
+	verIndex := strings.LastIndex(filename[:relIndex], "-")
+	ver = filename[verIndex+1 : relIndex]
+
+	epochIndex := strings.Index(filename, ":")
+	if epochIndex == -1 {
+		epoch = 0
+	} else {
+		epoch, _ = strconv.Atoi(filename[:epochIndex])
+	}
```

Comment:
For my understanding: What's `epochIndex` in the context of a filename? Do you have an example?

Reply:
You can see an example here.
https://github.com/aquasecurity/fanal/blob/15afdbeacb4bc40e101d9ef12ea1a7ab483ad7ce/analyzer/pkg/rpm/rpm.go#L100

`Epoch` is a number separated by a colon at the beginning of the package name.

Reply:
But unfortunately `SourceRpm` doesn't include `Epoch`. I don't know why, but I'm guessing it is the same as the epoch of the binary package.

Reply:
I see. thanks!